pkg/system/phase2_creating.go

```diff
@@ -845,97 +846,139 @@ func (r *Reconciler) UpgradeMigrateDB() error {
 	}
 
 	r.Logger.Infof("UpgradeMigrateDB: upgrade phase - %s", phase)
-	oldSts := &appsv1.StatefulSet{
+	mongoSts := &appsv1.StatefulSet{
 		TypeMeta: metav1.TypeMeta{Kind: "StatefulSet"},
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      "noobaa-db",
 			Namespace: options.Namespace,
 		},
 	}
-	if !util.KubeCheck(oldSts) {
-		r.Logger.Info("Old (mongo) STS was not found. skipping migration")
-		phase = nbv1.UpgradePhaseNone
-	} else {
-		replicas := int32(1)
-		if phase == nbv1.UpgradePhaseClean {
-			r.Logger.Info("UpgradeMigrateDB:: Cleanning phase - setting old STS to 0 replicas")
-			replicas = int32(0)
-		}
-		if err := r.ReconcileObject(oldSts, func() error { // remove old sts pods when finish migrating
-			oldSts.Spec.Replicas = &replicas
-			podSpec := &oldSts.Spec.Template.Spec
-			podSpec.ServiceAccountName = "noobaa"
-			for i := range podSpec.InitContainers {
-				c := &podSpec.InitContainers[i]
-				if c.Name == "init" {
-					c.Image = r.NooBaa.Status.ActualImage
-				}
-			}
-			return nil
-		}); err != nil {
-			r.Logger.Errorf("got error on mongo STS reconcile %v", err)
-			return err
-		}
-		replicas = int32(0)
-		if phase == nbv1.UpgradePhaseClean {
-			r.Logger.Info("UpgradeMigrateDB:: cleanning phase - setting endopint deployment to 1")
-			replicas = int32(1)
-		}
-		if err := r.ReconcileObject(r.DeploymentEndpoint, func() error { // remove endpoints pods before starting
-			r.DeploymentEndpoint.Spec.Replicas = &replicas
-			return nil
-		}); err != nil {
-			r.Logger.Errorf("got error on endpoints deployment reconcile %v", err)
-			return err
-		}
-		if phase == "" {
+
+	mongoExists := util.KubeCheck(mongoSts)
+
+	// if mongo STS exists (or we are in the middle of cleanup phase) then perform migration from mongo to postgres
+	if mongoExists || phase == nbv1.UpgradePhaseClean {
```

Comment:
addition due to deleting  right? Maybe this check is unneeded? like check in "" if we have mongo set to UpgradePhasePrepare and if we don't set to  UpgradePhaseNone. what do you think?

Reply:
in the cleaning phase, we still want to enter the switch after the STS is deleted

Reply:
Yeah I understood that, so wouldn't it be cleaner if we move it inside?

Reply:
ok. got you. I'll change it